[Bug]: aborted reply turns can stall for 30 seconds when delivery settlement starts after cancellation

### Bug type

Regression (worked before, now fails)

### Beta release blocker

No

### Summary

A reply turn whose delivery is stalled remains stuck in settlement for up to 30 seconds when its abort signal was already aborted before settlement begins.

### Steps to reproduce

1. Create a real reply dispatcher whose delivery promise does not settle.
2. Enqueue a final reply through the reply-turn ledger.
3. Abort an AbortController before passing its signal to the ledger's settlement method.
4. Race settlement against an already-resolved pending sentinel.

### Expected behavior

Settlement immediately resolves with its existing `aborted` outcome, matching the canonical abort-aware dispatcher idle waiter.

### Actual behavior

The boundary regression fails with `Expected: "aborted"; Received: "pending"`. The settlement method registers its abort listener after the abort event already fired, so the stalled dispatcher keeps settlement pending until its existing 30-second deadline; it then returns the same `aborted` outcome unnecessarily late.

### OpenClaw version

2026.8.1, source revision e294c154a6c7aace5ded3425587f191f13ce3e29

### Operating system

macOS 26.6.1

### Install method

Repository source checkout with its real reply dispatcher and Vitest owner-boundary regression.

### Model

Not applicable; this defect occurs after model execution in provider-independent reply settlement.

### Provider / routing chain

Not applicable; the reproduction directly exercises the provider-independent reply dispatcher and reply-turn ledger.

### Logs

```text
FAIL createReplyTurnLedger > settles immediately when the abort signal already fired
AssertionError: expected 'pending' to be 'aborted'
Expected: "aborted"
Received: "pending"
```

### Impact and severity

Aborted reply turns with stalled delivery unnecessarily delay finalization and idle cleanup for up to 30 seconds. The existing cancellation outcome, fallback policy, delivery visibility, replay ownership, and cleanup contracts do not need to change.

### Additional information

Introduced by settled-delivery receipt consolidation in #124773, merge commit eb38d5e4863a66d881c3fc1aa785ff53898088e7. The canonical sibling idle waiter already checks whether its signal was aborted before registering a listener.

--- a/src/auto-reply/reply/dispatch-from-config.turn-ledger.test.ts
+++ b/src/auto-reply/reply/dispatch-from-config.turn-ledger.test.ts
@@ -174,4 +174,27 @@ describe("createReplyTurnLedger", () => {
     dispatcher.markComplete();
     await dispatcher.waitForIdle();
   });
+
+  it("settles immediately when the abort signal already fired", async () => {
+    let releaseDeliver!: () => void;
+    const stalled = new Promise<void>((resolve) => {
+      releaseDeliver = resolve;
+    });
+    const dispatcher = createReplyDispatcher({ deliver: () => stalled });
+    const ledger = createReplyTurnLedger(dispatcher);
+    ledger.sendQueued("final", { text: "hello" });
+    const abortController = new AbortController();
+    abortController.abort();
+    const settled = ledger.settleQueued(abortController.signal);
+
+    try {
+      await expect(Promise.race([settled, Promise.resolve("pending")])).resolves.toBe("aborted");
+      expect(ledger.hasVisibleDelivery()).toBe(false);
+    } finally {
+      releaseDeliver();
+      dispatcher.markComplete();
+      await dispatcher.waitForIdle();
+      await settled;
+    }
+  });
 });

--- a/src/auto-reply/reply/dispatch-from-config.turn-ledger.ts
+++ b/src/auto-reply/reply/dispatch-from-config.turn-ledger.ts
@@ -101,6 +101,9 @@ export function createReplyTurnLedger(dispatcher: ReplyDispatcher): ReplyTurnLed
       }
     },
     async settleQueued(abortSignal) {
+      if (abortSignal?.aborted) {
+        return "aborted";
+      }
       let timedOut = false;
       let timer: ReturnType<typeof setTimeout> | undefined;
       const deadline = new Promise<void>((resolve) => {

